.pipelines/templates/validate-version.yml

```diff
@@ -0,0 +1,27 @@
+parameters:
+- name: UPSTREAM_BRANCH
+  type: string
+  default: remotes/origin/master
+steps:
+- task: Bash@3
+  displayName: 'verify version is changed'
+  name: verify
+  inputs:
+      targetType: 'inline'
+      script: |
+        set -eux  # exit on error
+    
+        GIT_DIFF_UPSTREAMBRANCH=${{ parameters.UPSTREAM_BRANCH }}
+        if [[ ! $GIT_DIFF_UPSTREAMBRANCH == *"/"* ]]; then
+          echo "missing full path, adding remotes/origin/"
+          GIT_DIFF_UPSTREAMBRANCH=remotes/origin/$GIT_DIFF_UPSTREAMBRANCH
+        fi
+
+        GIT_DIFF_SOURCEBRANCH="HEAD"
+
+        VERSION_FILECHANGE_SET=$(git diff "$GIT_DIFF_SOURCEBRANCH" "$GIT_DIFF_UPSTREAMBRANCH" --name-only | grep -w VERSION)
+        # check if file was changeds
+        if [ -z "$VERSION_FILECHANGE_SET" ]; then
+            echo "version file not changed"
+        fi
```

Comment:
should we add format validation with regex?

Reply:
maybe check that the version change does not change the major? (or even minor)

Reply:
Then how will we change it major eventually if not by PR ?

Reply:
agree with @SharonHart . adding validation logic here will be very strict.
the PRs are also going through manual review, this is just to "not miss" a version bump by mistake.
I will open a user story for validating the version bump in CI (@shiranr also mentioned this in her PR).

Reply:
added validation logic @SharonHart @shiranr @navalev :)

Reply:
looks good